packages/core/src/contracts/ownedAddressTCRWithAppeals.ts

```diff
@@ -0,0 +1,398 @@
+import { BigNumber } from "bignumber.js";
+import { Observable } from "rxjs";
+import "rxjs/add/operator/distinctUntilChanged";
+import "@joincivil/utils";
+import { ContentProvider } from "../content/contentprovider";
+import { CivilTransactionReceipt, EthAddress, TxHash, TxData } from "../types";
+import { requireAccount } from "../utils/errors";
+import { Web3Wrapper } from "../utils/web3wrapper";
+import { BaseWrapper } from "./basewrapper";
+import { OwnedAddressTCRWithAppealsContract } from "./generated/owned_address_t_c_r_with_appeals";
+import { Voting } from "./voting";
+import { EIP20 } from "./eip20";
+import { Parameterizer } from "./parameterizer";
+import { Newsroom } from "./newsroom";
+
+/**
+ * The OwnedAddressTCRWithAppeals tracks the status of addresses that have been applied and allows
+ * users to make transactions that modify the state of the TCR.
+ *
+ * Users can get stream of currently approved addresses, as well as specific information
+ * about individual listings (may or may not be approved)
+ *
+ * Users can also apply to the registry, challenge applications or listings, as well
+ * as collect winnings related to challenges, or withdraw/deposit tokens from listings.
+ */
+export class OwnedAddressTCRWithAppeals extends BaseWrapper<OwnedAddressTCRWithAppealsContract> {
```

Comment:
Abstract it into a nice API, I'd love to see some enum with phases and functions that accept/return phase instead of just huge spam of ifs that were necessary due to smart-contract. The goal here is to have a nice API for end-users, not 1:1 with smart-contracts.

And I imagine a lot less typing and easier refactoring later